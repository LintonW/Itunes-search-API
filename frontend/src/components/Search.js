import React, { useState } from 'react'
import Favourites from './Favourites';
import Landing from './Landing';

//search component with props
const Search = ({ handleSubmit, handleChange, handleChangeMedia, loaded, favouriteItems, deleteFavourite }) => {

    //useState
    const [favouritesLoaded, setFavouritesLoaded] = useState(false);
    
    //onClick function 
    const handleClickFavourites = () => setFavouritesLoaded(true);


    //rendering the component 
    return (
        <div className="search">
            <div className="search-contents">
                
                <h1>Itunes Store</h1>

                <form onSubmit={handleSubmit} action="">
                    <input onChange={handleChange} type="text"/>
                    
                    <div className="options">
                        <select onChange={handleChangeMedia}>
                            <option>Select An Option :</option>
                            <option value="all">All</option>
                            <option value="music">Music</option>
                            <option value="musicVideo">Music Video</option>
                            <option value="movie">Movies</option>
                            <option value="shortFilm">Short Film</option>
                            <option value="tvShow">Tv Show</option>
                            <option value="podcast">Podcast</option>
                            <option value="audiobook">Audiobook</option>
                            <option value="ebook">EBook</option>
                            <option value="software">Software</option>
                        </select>

                        <button>Search</button>
                        <button onClick={handleClickFavourites}>Favourites</button>
                        
                    </div>
                    
                </form>

            </div>

            {favouritesLoaded && 
                <Favourites 
                    favouriteItems={favouriteItems} 
                    loaded={loaded} 
                    deleteFavourite={deleteFavourite}/> 
            }

            {loaded ? '' : <Landing />}

           
            
        </div>
    )
}

export default Search
