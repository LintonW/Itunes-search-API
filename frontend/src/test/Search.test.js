import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../components/Search';

it('renders correctly', () => {
  const tree = renderer
    .create(<Search />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});