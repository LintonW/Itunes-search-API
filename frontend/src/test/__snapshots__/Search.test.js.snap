// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search"
>
  <div
    className="search-contents"
  >
    <h1>
      Itunes Store
    </h1>
    <form
      action=""
    >
      <input
        type="text"
      />
      <div
        className="options"
      >
        <select>
          <option>
            Select An Option :
          </option>
          <option
            value="all"
          >
            All
          </option>
          <option
            value="music"
          >
            Music
          </option>
          <option
            value="musicVideo"
          >
            Music Video
          </option>
          <option
            value="movie"
          >
            Movies
          </option>
          <option
            value="shortFilm"
          >
            Short Film
          </option>
          <option
            value="tvShow"
          >
            Tv Show
          </option>
          <option
            value="podcast"
          >
            Podcast
          </option>
          <option
            value="audiobook"
          >
            Audiobook
          </option>
          <option
            value="ebook"
          >
            EBook
          </option>
          <option
            value="software"
          >
            Software
          </option>
        </select>
        <button>
          Search
        </button>
        <button
          onClick={[Function]}
        >
          Favourites
        </button>
      </div>
    </form>
  </div>
  <div
    className="landing-section"
  >
    <img
      alt="searchimg"
      src="search.svg"
    />
  </div>
</div>
`;
